pandas/tseries/index.py

```diff
@@ -1629,13 +1632,17 @@ def normalize(self):
         return DatetimeIndex(new_values, freq='infer', name=self.name,
                              tz=self.tz)
 
-    def searchsorted(self, key, side='left'):
-        if isinstance(key, (np.ndarray, Index)):
-            key = np.array(key, dtype=_NS_DTYPE, copy=False)
+    @Substitution(klass='DatetimeIndex')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
+        validate_args(args, min_length=0, max_length=1)
```

Comment:
easy, ``out`` is the user expecting an in-place modification, which we don't do that way, so we MUST raise (prob ``ValueError``). while ``sorter`` we don't care whether its passed or not and so it gets passed thru to satisfy numpy compat.